Cognite.Simulator.Utils/DefaultClasses/DefaultConnectorRuntime.cs

```diff
@@ -51,7 +60,138 @@ public static async Task RunStandalone()
                 }
             }
         }
+    }
+
+
+    private static bool AreCollectionsEqual<T>(IEnumerable<T> col1, IEnumerable<T> col2) where T : class
+    {
+        if (col1 == null || col2 == null)
+            return col1 == col2; // Both null is true, one null is false
+
+        // Convert collections to lists for easier manipulation
+        var list1 = col1.ToList();
+        var list2 = col2.ToList();
+
+        if (list1.Count != list2.Count)
+            return false;
 
+        // Compare each item in the collections
+        for (int i = 0; i < list1.Count; i++)
+        {
+            if (!AreObjectsEqual(list1[i], list2[i]))
+                return false;
+        }
+
+        return true;
+    }
+
+    private static bool AreObjectsEqual<T>(T obj1, T obj2) where T : class
+    {
+        if (obj1 == null || obj2 == null)
+            return obj1 == obj2; // Both null is true, one null is false
+
+        var properties = typeof(T).GetProperties();
+
+        foreach (var property in properties)
+        {
+            if (property.CanWrite) {
+                var value1 = property.GetValue(obj1);
+                var value2 = property.GetValue(obj2);
+                if (property.PropertyType == typeof(IEnumerable<SimulatorUnitEntry>)) {
+                    return AreObjectsEqual(value1, value2);
+                } else if (property.PropertyType == typeof(IEnumerable<SimulatorStepFieldParam>)) {
+                    return AreObjectsEqual(value1, value2);
+                }{
+                    if (!object.Equals(value1, value2)){
+                        return false;
+                    }
+                }
+            }
+        }
+        return true;
+    }
+
+    /// <summary>
+    /// An method for creating a new simulator definition if anything was changed in the existing one
+    /// </summary>
+    /// <param name="existingSimulatorDefinition"></param>
+    /// <param name="newSimulatorDefinition"></param>
+    /// <param name="logger"></param>
+    /// <returns></returns>
+    /// <exception cref="Exception"></exception>
+    public static SimulatorUpdateItem CreateUpdateItemIfChanged(Simulator existingSimulatorDefinition, 
+    SimulatorCreate newSimulatorDefinition, ILogger<DefaultConnectorRuntime<TAutomationConfig, TModelState, TModelStateBasePoco>> logger )
+    {
+        if (existingSimulatorDefinition == null) {
+            throw new Exception("Simulator definition (from CDF) is null");
```

Comment:
```suggestion
            throw new Exception("Remote simulator definition is null");
```
all this code is CDF, better not say CDF. If strictly necessary, the full name "Cognite Data Fusion" is preferred.